File: check_db.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const dbPath = path.join(process.env.APPDATA, 'com.aszeromo.smarticafe', 'smarticafe.db');
console.log('Checking database at:', dbPath);

const db = new sqlite3.Database(dbPath, sqlite3.OPEN_READONLY, (err) => {
  if (err) {
    console.error('Error opening database:', err.message);
    process.exit(1);
  }
});

db.serialize(() => {
  db.get("SELECT COUNT(*) as count FROM auth_accounts WHERE role='admin'", [], (err, row) => {
    if (err) {
      console.error('Error querying auth_accounts:', err.message);
    } else {
      console.log('ADMIN_COUNT:', row.count);
      if (row.count === 0) {
        console.log('STATUS: BOOTSTRAP_REQUIRED');
      } else {
        console.log('STATUS: READY');
      }
    }
  });

  db.all("SELECT k, v FROM kv", [], (err, rows) => {
    if (err) {
      console.error('Error querying kv:', err.message);
    } else {
      console.log('KV_SETTINGS:', rows);
    }
    db.close();
  });
});

File: check_db.py
import sqlite3
import os

db_path = os.path.join(os.environ['APPDATA'], 'com.aszeromo.smarticafe', 'smarticafe.db')
print(f"Checking database: {db_path}")

if not os.path.exists(db_path):
    print("Database file not found!")
    exit(0)

try:
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    # Check admin count
    cursor.execute("SELECT COUNT(*) FROM auth_accounts WHERE role='admin'")
    admin_count = cursor.fetchone()[0]
    print(f"ADMIN_COUNT: {admin_count}")
    
    # Check KV settings
    cursor.execute("SELECT k, v FROM kv")
    settings = cursor.fetchall()
    print(f"KV_SETTINGS: {settings}")
    
    conn.close()
except Exception as e:
    print(f"Error: {e}")
